awx_collection/test/awx/test_job_template.py

```diff
@@ -48,11 +48,19 @@ def test_job_launch_with_prompting(run_module, admin_user, project, inventory, m
     result = run_module('tower_job_launch', dict(
         job_template='foo',
         inventory=inventory.name,
-        credential=machine_credential.name
+        credential=machine_credential.name,
+        extra_vars={"var1": "My First Variable",
+                    "var2": "My Second Variable",
+                    "var3": "My Third Variable"
+                    }
     ), admin_user)
     assert result.pop('changed', None), result
 
     job = Job.objects.get(id=result['id'])
+    assert result['invocation']['module_args']['extra_vars'] == {'var1': 'My First Variable',
+                                                                 'var2': 'My Second Variable',
+                                                                 'var3': 'My Third Variable'
+                                                                 }
```

Comment:
this is actually somewhat of a weak assertion. If it failed to take effect, it would still pass. You could change the JT to ask_variables_on_launch, and then check `job.extra_vars`.